### ISSUE TITLE
Compilation error in generated code when table valued function returns table with client side computed columns and <recordsImplementingRecordN/> is active

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#14830]
The demo (https://github.com/jOOQ/demo) has a use-case where client side computed columns are added to the `CUSTOMER` table, producing `FULL_ADDRESS` and `FULL_NAME` columns, but those columns aren't produced by the `REWARDS_REPORT` table valued function, which returns `CustomerRecord` as well, since jOOQ 3.18: #5405.

The problem only appears when `<recordsImplementingRecordN/>` is active, in case of which we're getting:

```java
public class CustomerRecord extends UpdatableRecordImpl<CustomerRecord> implements Record12<Long, Long, String, String, String, Long, Boolean, LocalDate, LocalDateTime, Integer, String, String> {
```

But:

```java
public class RewardsReport extends TableImpl<CustomerRecord> {

    // [...]

    public <U> SelectField<U> mapping(Function10<? super Long, ? super Long, ? super String, ? super String, ? super String, ? super Long, ? super Boolean, ? super LocalDate, ? super LocalDateTime, ? super Integer, ? extends U> from) {
        return convertFrom(Records.mapping(from));
    }
```

The 2 extra columns from the `CustomerRecord` aren't being generated.

**Comment by lukaseder** [comment]
It's not sure yet whether table valued functions should always inherit all synthetic properties of their base tables, such as foreign keys, see https://github.com/jOOQ/jOOQ/issues/14831

In any case, table valued functions and base tables can be configured independently in the code generator, meaning their row type can deviate.

As such, the bug here is that these `mapping()` convenience methods must not be generated if the row type is not identical (degree and column names!). It would be tempting to derive a mapping that curries functions, but there are many ways we can get this wrong, so let's just avoid generating the convenience method for now.

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.19.0 and 3.18.2 (https://github.com/jOOQ/jOOQ/issues/14832)
